jellyfin_helpers/jellyfin_helpers/parse_library.py

```diff
@@ -0,0 +1,116 @@
+"""Minimal import example main file."""
+import argparse
+import os
+import pathlib
+import sys
+from collections import defaultdict
+
+import pandas as pd
+
+music_extensions = [".mp3", ".m4a", ".wav", ".flac", ".ogg", ".mp4", ".mid"]
+
+
+# def import_module_from_file(file_path: str, module_name: str = "cfg"):
+#     return importlib.machinery.SourceFileLoader(
+#         module_name, file_path
+#     ).load_module()
+
+
+def parse_cmd_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--library_path",
+        "-l",
+        help="Path to top level of libary to scan.",
+        required=True,
+    )
+    args = parser.parse_args()
+    return args
+
+
+def get_dir_stats(path_str: str, recursive: bool = False) -> {}:
+
+    if recursive:
+        sub_paths = list(path_str.rglob("*"))
+    else:
+        sub_paths = list(path_str.glob("*"))
+
+    # not defaultdict, need to explicitly set!
+    result_dict = {"n_dirs": 0, "n_files": 0, "n_music": 0, "n_other": 0}
+    for sub_path in sub_paths:
+        if sub_path.exists():  # should be redundant, keep for safety
+            if sub_path.is_dir():
+                result_dict["n_dirs"] += 1
+            else:
+                result_dict["n_files"] += 1
+                if os.path.splitext(sub_path)[-1] in music_extensions:
+                    result_dict["n_music"] += 1
+                else:
+                    result_dict["n_other"] += 1
+
+    return result_dict
+
+
+def get_library_as_df(
+    root_path: pathlib.Path,
+    depth: int = 1,
+    max_depth: int = 2,
+    limit: int = None,
+) -> pd.DataFrame:
+
+    if (max_depth < 1) or (depth > max_depth):
```

Comment:
These if-statements do not make sense to me. What is the goal of them? They would be clearer if they were in a method or had a comment explaining why they were needed.